### ISSUE TITLE
Cannot create LDAP user via /api/v4/users, error about a password that is not needed

### ISSUE THREAD

**Issue by towolf** [mattermost/mattermost-server#12845]
#### Summary
We have been using a script to create LDAP users ahead of time before they log in the first time, so that we can join them to the appropriate teams and channels already.

This is using the Mattermost API.

This stopped working apparently in 5.15.0 and is still present in 5.16.0.

Instead of creating the user successfully, where no password should be provided since LDAP handles this, the error that is returned is `"Your password must contain at least 10 characters made up of at least one lowercase letter, at least one uppercase letter, at least one number, and at least one symbol (e.g. \"~!@#$%^&*()\")."`.

#### Steps to reproduce

```
# curl -s -XPOST -H "Authorization: Bearer pmab...qj5na" https://mattermost-dev.example.com/api/v4/users  -d '{"email": "foo.moo@b.c", "username": "nope", "auth_service": "ldap", "auth_data": "999099"}' | jq
{
  "id": "model.user.is_valid.pwd.app_error",
  "message": "Your password must contain at least 10 characters made up of at least one lowercase letter, at least one uppercase letter, at least one number, and at least one symbol (e.g. \"~!@#$%^&*()\").",
  "detailed_error": "",
  "request_id": "aqfmmurqzjbp7nzuwgswysostc",
  "status_code": 400
}
```

#### Expected behavior

But no password complexity policy should apply here, since I specified no password!?!?

This is covered in the docs: 
>  Password is required for email login. For other authentication types such as LDAP or SAML, auth_data and auth_service fields are required.

https://api.mattermost.com/#tag/users%2Fpaths%2F~1users%2Fpost

**Comment by amyblais** [comment]
@towolf One of our devs reproduced this and is looking into this. I will find out if a bug ticket was opened for this and will link it here.

**Comment by amyblais** [comment]
Bug ticket here: https://mattermost.atlassian.net/browse/MM-19766.

**Comment by amyblais** [comment]
This is fixed for upcoming v5.18 release which will release on December 16th,
